activerecord/test/cases/bind_parameter_test.rb

```diff
@@ -30,6 +32,11 @@ def setup
     end
 
     if ActiveRecord::Base.connection.supports_statement_cache?
+      def test_bind_from_join_in_subquery
+        scope = Author.from(Author.joins(:thinking_posts).where(id: 1), 'authors').where(id: 1)
```

Comment:
This test won't fail if the order is wrong (and in fact I'm pretty sure it is wrong) since the where statements are identical.